File: backend/app/api/v1/password.py
from fastapi import APIRouter

from app.schemas.password import GenerateRequest, GenerateResponse, ValidateRequest, ValidateResponse
from app.services.password_generator import generate_password, generate_batch
from app.services.password_validator import validate_password

router = APIRouter()


@router.post("/generate", response_model=GenerateResponse)
async def generate(body: GenerateRequest):
    passwords = generate_batch(
        count=body.count,
        length=body.length,
        include_uppercase=body.include_uppercase,
        include_digits=body.include_digits,
        include_special=body.include_special,
        exclude_similar=body.exclude_similar,
        exclude_ambiguous=body.exclude_ambiguous,
    )
    return GenerateResponse(passwords=passwords)


@router.post("/validate", response_model=ValidateResponse)
async def validate(body: ValidateRequest):
    result = validate_password(body.password)
    return ValidateResponse(**result)

File: frontend/src/api/password.ts
import api from './index'

export const passwordApi = {
  generate(params: {
    length?: number
    include_uppercase?: boolean
    include_digits?: boolean
    include_special?: boolean
    exclude_similar?: boolean
    exclude_ambiguous?: boolean
    count?: number
  }) {
    return api.post('/password/generate', params)
  },
  validate(password: string) {
    return api.post('/password/validate', { password })
  },
}
